sigstore/sign.py

```diff
@@ -313,9 +314,14 @@ class SigningResult(BaseModel):
     Represents the artifacts of a signing operation.
     """
 
+    digest_algorithm: HashAlgorithm
+    """
+    The digest algorithm to use for the hash.
+    """
+
     input_digest: HexStr
     """
-    The hex-encoded SHA256 digest of the input that was signed for.
+    The hex-encoded digest of the input that was signed for.
     """
```

Comment:
NB: Both of these fields will disappear with #804 and we intend to remove `SigningResult` entirely with #854, so we should probably avoid more fields here. Given that `HashAlgorithm` will currently always be invariant, maybe we can just hardcode it with a `NOTE` for now?

Reply:
as soon as we update rekor (probably next week), it will no longer be an invariant, ie there wil be 2 supported hashes (sha256 and some dirhash hash)

Reply:
Gotcha -- in that case, we should probably bump up the priority on #854, since `SigningResult` has outlived its utility in the world of Sigstore bundles 🙂

Reply:
SGTM. So you'd land https://github.com/sigstore/sigstore-python/issues/854 then this PR I suppose? Or you don't really care?

Reply:
Tangent, but is dirhash documented anywhere? I see there's a dirhash standard on GitHub but I wasn't sure if it was the same thing.

(Part of what I'm wondering is what the mapping between a dirhash and a standard algorithm is -- how do we form a `Prehashed` for them?)

Reply:
That matches my understanding, thanks! We should probably make sure that subclassing `HashAlgorithm` like that for external hashes is okay in terms of PyCA's APIs, but otherwise that seems reasonable to me.